### PR TITLE
pyTorch integration with simple model loader

### PR DESCRIPTION
**Pull request by Arslan-Siraj**
## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by Arslan-Siraj** [comment]
Successfully import prototype pytorch models via CMake in OpenNMS.
Next will import alphapeptdeep, for real case.

File: src/openms_ml/include/OpenMS/ML/OpenMSMLWrapper.h [hunk 1]
```diff
@@ -32,8 +32,76 @@
...
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg $
+// $Authors: Timo Sachsenberg $
+// --------------------------------------------------------------------------
+
 #pragma once
 
+#include <OpenMS/ML/OpenMS_MLConfig.h>
+#include <torch/torch.h>
```
Comment (timosachsenberg) [inline src/openms_ml/include/OpenMS/ML/OpenMSMLWrapper.h:72]: Would be good if we would not have to include torch headers here. 
for the model variable we can use the PIMPL pattern. For the type in the vector I don’t know yet. But it should be a normal e.g. int not a torch typedef. Then this header would not need to include torch and thus not expose the torch symbols/header to the rest of OpenMS

Comment (timosachsenberg) [reply src/openms_ml/include/OpenMS/ML/OpenMSMLWrapper.h:72]: E.g. pass vector with double or int and use in cpp file: https://stackoverflow.com/questions/73229831/how-to-populate-torchjitivalue-with-c-array-without-using-torchfrom-blob

**Comment by timosachsenberg** [comment]
I think I will close this in favour of https://github.com/OpenMS/OpenMS/pull/6784.
Feel free to reopen if I missed something.

**Pull request closed without merging**